blueocean-dashboard/src/main/js/components/testing/TestResults.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { Component, PropTypes } from 'react';
+import { ResultItem, StatusIndicator } from '@jenkins-cd/design-language';
+import moment from 'moment';
+
+/* eslint-disable max-len */
+
+const TestCaseResultRow = (props) => {
+    const t = props.testCase;
+    const duration = moment.duration(Number(t.duration), 'milliseconds').humanize();
+    const expandable = t.errorStackTrace;
+
+    let testDetails = !expandable ? null : (<div className="test-details">
+        <div className="test-detail-text" style={{ display: 'none' }}>
+            {duration}
+        </div>
+        <div className="test-console">
+            <h4>Error</h4>
+            <div className="error-message">
+                {t.errorDetails}
+            </div>
+            <h4>Output</h4>
+            <div className="stack-trace">
+                {t.errorStackTrace}
+            </div>
+        </div>
+    </div>);
+    
+    let statusIndicator = null;
+    switch (t.status) {
+    case 'FAILED':
+        statusIndicator = StatusIndicator.validResultValues.failure;
+        break;
+    case 'SKIPPED':
+        statusIndicator = StatusIndicator.validResultValues.unstable;
+        break;
+    case 'PASSED':
+        statusIndicator = StatusIndicator.validResultValues.success;
+        break;
+    default:
+    }
+    
+    return (<ResultItem
+      result={statusIndicator}
+      expanded={false}
+      label={`${t.name} - ${t.className}`}
+      onExpand={null}
+      extraInfo={duration}
+    >
+        { testDetails }
+    </ResultItem>);
+};
+
+TestCaseResultRow.propTypes = {
+    testCase: PropTypes.object,
+};
+
+export default class TestResult extends Component {
+
+    render() {
+        const testResults = this.props.testResults;
+        const suites = this.props.testResults.suites;
+        const tests = [].concat.apply([], suites.map(t => t.cases));
+        
+        // possible statuses: PASSED, FAILED, SKIPPED
+        const failures = tests.filter(t => t.status === 'FAILED');
+        const skipped = tests.filter(t => t.status === 'SKIPPED');
+        const newFailures = failures.filter(t => t.age === 1);
+        const existingFailures = failures.filter(t => t.age > 1);
+
+        let passBlock = null;
+        let newFailureBlock = null;
+        let existingFailureBlock = null;
+        let skippedBlock = null;
+        
+        if (testResults.failCount === 0) {
+            passBlock = [
+                <h4>Passing - {testResults.passCount}</h4>,
+                suites.map((t, i) => <TestCaseResultRow key={i} testCase={{
+                    className: `${t.cases.filter(c => c.status === 'PASSED').length} Passing`, // this shows second
+                    name: t.name,
+                    duration: t.duration,
+                    status: 'PASSED',
+                }} />),
+            ];
+        }
+
+        if (newFailures.length > 0 || existingFailures.length > 0) {
```

Comment:
No, not used. But I want to show the `newFailures` block in the case there are new failures or existing failures, to make it clear if a given build has no _new_ failures.